### PR TITLE
Difhel refactor

### LINKED ISSUE #15: Fix English translation in the cli interface

### LINKED ISSUE #16: Don't use `username` as constant key; use `id` instead
username - изменяемое поле, его **нельзя** использовать как ключ!
используй _id

_Originally posted by @difhel in https://github.com/elansteam/backend/pull/11#discussion_r1407353805_

### PR DESCRIPTION
**Pull request by difhel**
Here are some updates made in this branch:
- removed MongoDB ObjectID from all models
- ids of `User` and `Group` are now integers; ids of `GroupRole`, `Role` and `Entity` (domain) are now strings
- `GroupRole` id looks like that: `group<associated_group_id>_<local_group_name>`
- added `mypy` and `ruff` linting (run `python3 manage.py lint`)
- now logger in the CLI shows the name of the running linter
- refactored CLI: added subparses, better documentation (`python3 manage.py -h`), `--config` parameter is now required, but only for `lint` and `run` command
- CLI command `runserver` renamed to `run`
- `pytest`, `pylint`, `ruff` and `mypy` dependencies moved to dev dependencies in `pyproject.toml` (poetry configuration file) - they are not necessary for running server
- created ~`NameRouter`~ `Domain Router` (`NameRouter` was renamed)
- wrote module `utils.regex_validators` for email validation
- new logic in sign in method: user can use his username (e.g. `scintilla_aaa`), id (e.g. `id42`) or email (e.g. `test@difhel.dev`)
- all submodules now have well-formed `__init__.py` files (this is step following PEP-8 and fixing `mypy`-detected issues)
- many methods in the database models were renamed
- deleted adressing by names (domains) - now using only ids (closes #16 )
- fixed #15 
- now CLI `run` command crashes with error description if user provides example.env as his config

### CODE REVIEW

**Comment by difhel** [comment]
Please note that these changes are breaking and they should be fixed before merging with master.
**This is significantly important.**

But I suggest not to implement testing with `pytest` in this pull request and implement that later.

File: src/db/managers/user_database_manager.py [hunk 1]
```diff
@@ -1,59 +1,74 @@
...
         """
 
-        await self.db.update_one({"name": user_name},
-                                 {"$push": {"roles": role_name}})
+        await self.collection.update_one(
+            {"_id": user_id},
+            {"$push": {"roles": role_id}}
+        )
+
+    async def delete_role(self, user_id: int, role_id: str) -> None:
```
Comment (Spotika) [inline src/db/managers/user_database_manager.py:56]: Протестировать это

File: src/routers/auth_router.py [hunk 2]
```diff
@@ -1,33 +1,80 @@
...
+
+
+@router.post(
+    "/signup",
+    response_model=get_response_model(User),
+    responses={
+        400: get_error_schema("Signup failed")
     }
+)
+async def signup(user: UserSignup):
```
Comment (Spotika) [inline src/routers/auth_router.py:55]: В этом методе отсутствует удаление сущности, если при создании пользователя произошла ошибка

File: src/routers/roles_router.py [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+        _current_user: current user which created the role
+    Returns:
+        Created role
+    """
+    role_id = role_name_to_id(role.name)
 
-    if await db.role.get_by_name(role.name) is not None:
-        return get_error_response(f"Role with name <{role.name}> doesn't exist")
+    if await db.role.get(role_id) is not None:
+        return get_error_response(f"Role with id <{role_id}> already exists")
```
Comment (Spotika) [inline src/routers/roles_router.py]: Сюда нужно добавить логирование и вместо такой ошибки сделать статус в upper case

**Review by Spotika — CHANGES_REQUESTED** [review]

File: src/db/managers/user_database_manager.py (hunk 1 above)
Comment (Spotika) [reply src/db/managers/user_database_manager.py:56]: Так же отсутствует метод в api который предоставляет к этому доступ

**Review by Spotika — DISMISSED** [review]
It works and looks pretty good

File: src/auth/utils.py [hunk 4]
```diff
@@ -40,6 +23,9 @@ def has_role_permissions(role_staff: int, *permissions: Permissions) -> bool:
     Returns:
         True - if role has permission, else False
     """
+    if role_staff % 2 == 1:
```
Comment (difhel) [inline src/auth/utils.py]: TODO: delete $2^0$

File: src/auth/utils.py [hunk 5]
```diff
@@ -2,34 +2,31 @@
...
-    CAN_CREATE_USER = 1
-    CAN_SET_ROLE = 2
-    CAN_CREATE_ROLE = 3
-    CAN_ADD_USER_TO_GROUP = 4
-    CAN_CREATE_GROUP = 5
-    CAN_CREATE_GROUP_ROLE = 6
-    CAN_ADD_GROUP_ROLE = 7
-    CAN_ADD_ROLE_TO_USER = 8
-    # TODO: add more perms
+class AuthException(Exception):
```
Comment (difhel) [inline src/auth/utils.py:17]: TODO: fix _naming in AuthException (shadow name out of scope)

File: src/config.py [hunk 6]
```diff
@@ -1,29 +1,62 @@
...
+from loguru import logger
 
 
 class Config:
     """Entire project config"""
-
-    db_connect_url: str = os.environ.get("DB_CONNECT_URL")
-
-    db_name: str = os.environ.get("DB_NAME")
+    if any(arg_value is None for arg_value in [
```
Comment (difhel) [inline src/config.py:9]: Сделать бананы адекватными

File: src/db/__init__.py [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
-from db.managers.role_database_manager import RoleDatabaseManager
 from db.managers.group_database_manager import GroupDatabaseManager
+from db.managers.domain_router_database_manager import DomainRouterDatabaseManager
+from db.managers.role_database_manager import RoleDatabaseManager
 
-user = UserDatabaseManager()
-group = GroupDatabaseManager()
-group_role = GroupRoleDatabaseManager()
-role = RoleDatabaseManager()
+user: UserDatabaseManager = UserDatabaseManager()
```
Comment (difhel) [inline src/db/__init__.py:7]: Remove typings

File: src/db/models/role.py [hunk 8]
```diff
@@ -1,31 +1,61 @@
...
-        Args:
-            value: permission code
-        Raises:
-            ValidationError: if role code not positive
-        """
-        if value < 0:
-            raise ValidationError("Role code must be positive")
-        return value
+
+def role_name_to_id(role_name: str) -> str:
```
Comment (difhel) [inline src/db/models/role.py:53]: ID роли должен назначаться пользователем

**Review by difhel — COMMENTED** [review]
как там это... lgbt

**Review by difhel — COMMENTED** [review]
lgtm

**Review by Spotika — APPROVED** [review]
​  ⡏⠉⠉⠉⠉⠉⠉⠋⠉⠉⠉⠉⠉⠉⠋⠉⠉⠉⠉⠉⠉⠉⠉⠉⠉⠙⠉⠉⠉⠹ 
   ⡇⢸⣿⡟⠛⢿⣷⠀⢸⣿⡟⠛⢿⣷⡄⢸⣿⡇⠀⢸⣿⡇⢸⣿⡇⠀⢸⣿⡇⠀ 
 ​  ⡇⢸⣿⣧⣤⣾⠿⠀⢸⣿⣇⣀⣸⡿⠃⢸⣿⡇⠀⢸⣿⡇⢸⣿⣇⣀⣸⣿⡇⠀ 
 ​  ⡇⢸⣿⡏⠉⢹⣿⡆⢸⣿⡟⠛⢻⣷⡄⢸⣿⡇⠀⢸⣿⡇⢸⣿⡏⠉⢹⣿⡇⠀ 
 ​  ⡇⢸⣿⣧⣤⣼⡿⠃⢸⣿⡇⠀⢸⣿⡇⠸⣿⣧⣤⣼⡿⠁⢸⣿⡇⠀⢸⣿⡇⠀ 
 ​  ⣇⣀⣀⣀⣀⣀⣀⣄⣀⣀⣀⣀⣀⣀⣀⣠⣀⡈⠉⣁⣀⣄⣀⣀⣀⣠⣀⣀⣀⣰ 
 ​  ⣇⣿⠘⣿⣿⣿⡿⡿⣟⣟⢟⢟⢝⠵⡝⣿⡿⢂⣼⣿⣷⣌⠩⡫⡻⣝⠹⢿⣿⣷ 
 ​  ⡆⣿⣆⠱⣝⡵⣝⢅⠙⣿⢕⢕⢕⢕⢝⣥⢒⠅⣿⣿⣿⡿⣳⣌⠪⡪⣡⢑⢝⣇ 
 ​  ⡆⣿⣿⣦⠹⣳⣳⣕⢅⠈⢗⢕⢕⢕⢕⢕⢈⢆⠟⠋⠉⠁⠉⠉⠁⠈⠼⢐⢕⢽ 
 ​  ⡗⢰⣶⣶⣦⣝⢝⢕⢕⠅⡆⢕⢕⢕⢕⢕⣴⠏⣠⡶⠛⡉⡉⡛⢶⣦⡀⠐⣕⢕ 
 ​  ⡝⡄⢻⢟⣿⣿⣷⣕⣕⣅⣿⣔⣕⣵⣵⣿⣿⢠⣿⢠⣮⡈⣌⠨⠅⠹⣷⡀⢱⢕ 
 ​  ⡝⡵⠟⠈⢀⣀⣀⡀⠉⢿⣿⣿⣿⣿⣿⣿⣿⣼⣿⢈⡋⠴⢿⡟⣡⡇⣿⡇⡀⢕ 
 ​  ⡝⠁⣠⣾⠟⡉⡉⡉⠻⣦⣻⣿⣿⣿⣿⣿⣿⣿⣿⣧⠸⣿⣦⣥⣿⡇⡿⣰⢗⢄ 
 ​  ⠁⢰⣿⡏⣴⣌⠈⣌⠡⠈⢻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣬⣉⣉⣁⣄⢖⢕⢕⢕ 
 ​  ⡀⢻⣿⡇⢙⠁⠴⢿⡟⣡⡆⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⣵⣵⣿ 
 ​  ⡻⣄⣻⣿⣌⠘⢿⣷⣥⣿⠇⣿⣿⣿⣿⣿⣿⠛⠻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿ 
 ​  ⣷⢄⠻⣿⣟⠿⠦⠍⠉⣡⣾⣿⣿⣿⣿⣿⣿⢸⣿⣦⠙⣿⣿⣿⣿⣿⣿⣿⣿⠟ 
 ​  ⡕⡑⣑⣈⣻⢗⢟⢞⢝⣻⣿⣿⣿⣿⣿⣿⣿⠸⣿⠿⠃⣿⣿⣿⣿⣿⣿⡿⠁⣠ 
 ​  ⡝⡵⡈⢟⢕⢕⢕⢕⣵⣿⣿⣿⣿⣿⣿⣿⣿⣿⣶⣶⣿⣿⣿⣿⣿⠿⠋⣀⣈⠙ 
 ​  ⡝⡵⡕⡀⠑⠳⠿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠿⠛⢉⡠⡲⡫⡪⡪⡣
~~lgbt~~ 
**LGTM**

File: src/auth/utils.py (hunk 4 above)
Comment (Spotika) [reply src/auth/utils.py]: Fixed now

**Pull request merged**